src/test/java/org/openelisglobal/integrationTests/referenceTables/ReferenceTablesServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package org.openelisglobal.integrationTests.referenceTables;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openelisglobal.BaseTestConfig;
+import org.openelisglobal.config.TestConfig;
+import org.openelisglobal.referencetables.service.ReferenceTablesService;
+import org.openelisglobal.referencetables.valueholder.ReferenceTables;
+import org.openelisglobal.testUtils.ObjectsInitiator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = { BaseTestConfig.class, TestConfig.class })
+@TestPropertySource("classpath:common.properties")
+@ActiveProfiles("test")
+public class ReferenceTablesServiceTest {
+
+    @Autowired
+    private ReferenceTablesService referenceTablesService;
+
+    private String id;
+
+    private ReferenceTables expected;
+    private ReferenceTables result;
+
+
+    @Test
+    public void insert_shouldInsertReferenceTable(){
+        expected = ObjectsInitiator.createReferenceTable("X-Table","N","Y");
+        id = referenceTablesService.insert(expected);
+
+        result = referenceTablesService.get(id);
+        assertEquals(expected.getIsHl7Encoded(),result.getIsHl7Encoded());
+
+    }
+
+    @Test
+    public void update_shouldUpdateReferenceTable(){
+
+        expected = ObjectsInitiator.createReferenceTable("New Table","N","Y");
+
+        expected.setTableName("Update Table");
+        expected.setIsHl7Encoded("Y");
+        expected.setKeepHistory("N");
+
+        id = referenceTablesService.insert(expected);
+        expected = referenceTablesService.get(id);
+
+        result = referenceTablesService.update(expected);
+
+        assertEquals(expected.getTableName(), result.getTableName());
+        assertEquals(expected.getKeepHistory(), result.getKeepHistory());
+        assertEquals(expected.getIsHl7Encoded(), result.getIsHl7Encoded());
+        assertEquals(expected.getId(),result.getId());
```

Comment:
can you move the changes (line 49-51) so that they happen in between the insert and the update, so we can test that the object does get successfully updated? 

maybe test
- insert New table
- assert insert success as is
- change values
- call update
- assert update success on result

Then maybe create second method that does all the same steps except for the last one and instead:
- get updated values by id
- assert success on result of get
- assert success that result of get = result of update

This would enable us to see that the result returned by the update is the same as what is saved to what can be fetched from the database